model/src/main/kotlin/Defect.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model
+
+import com.fasterxml.jackson.annotation.JsonInclude
+
+import java.net.URI
+import java.time.Instant
+
+/**
+ * A data model for software defects.
+ *
+ * Instances of this class are created by advisor implementations that retrieve information about known defects in
+ * packages.
+ */
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+data class Defect(
+    /**
+     * The (external) ID of this defect. This is a string used by a concrete issue tracker system to reference this
+     * defect, such as a bug ID or ticket number.
+     */
+    val id: String,
+
+    /**
+     * The URL pointing to the source of this defect. This is typically a reference into the issue tracker system that
+     * contains this defect.
+     */
+    val url: URI,
+
+    /**
+     * A title for this defect if available. This is a short summary describing the problem at hand.
+     */
+    val title: String? = null,
+
+    /**
+     * A state of the associated defect if available. The concrete meaning of this string depends on the source from
+     * where it was obtained, as different issue tracker systems use their specific terminology. Possible values could
+     * be *OPEN*, *IN PROGRESS*, *BLOCKED*, etc.
+     */
+    val state: String? = null,
+
+    /**
+     * The severity assigned to the defect if available. The meaning of this string depends on the source system.
+     */
+    val severity: String? = null,
```

Comment:
I cannot find my previous comment on this anymore. Using String is quite generic. 
This implies to me that there are no plans to use the severity in ORT. Is this the case?

Reply:
> 
> 
> I cannot find my previous comment on this anymore. Using String is quite generic.
> This implies to me that there are no plans to use the severity in ORT. Is this the case?

Yes, it is currently not planned to reuse ORT's severity, as this would imply some mapping logic, which might become rather complex. In the vulnerabilities model, there is also currently no unification of the several severity scores; we just provide the severity as a string and the scoring system it belongs to; I think, the situation is similar here.